pkg/asset/machines/azure/machines.go

```diff
@@ -110,15 +110,17 @@ func provider(platform *azure.Platform, mpool *azure.MachinePool, osImage string
 		image.Offer = mpool.OSImage.Offer
 		image.SKU = mpool.OSImage.SKU
 		image.Version = mpool.OSImage.Version
-	} else {
+	} else if useImageGallery {
 		// image gallery names cannot have dashes
 		galleryName := strings.Replace(clusterID, "-", "_", -1)
 		id := clusterID
 		if hyperVGen == "V2" {
 			id += "-gen2"
 		}
-		imageID := fmt.Sprintf("/resourceGroups/%s/providers/Microsoft.Compute/galleries/gallery_%s/images/%s/versions/%s", rg, galleryName, id, rhcosVersion)
+		imageID := fmt.Sprintf("/resourceGroups/%s/providers/Microsoft.Compute/galleries/gallery_%s/images/%s/versions/latest", rg, galleryName, id)
```

Comment:
Does using `latest` here work?

Reply:
@Prashanth684 I remember you had issues when using anything other than an actual version here

Reply:
yes it did - looking back at this comment: https://github.com/openshift/installer/pull/6304#discussion_r968799213, it did not like `latest` when used as a version

Reply:
See `name` here https://learn.microsoft.com/en-us/azure/templates/microsoft.compute/2021-10-01/galleries/images/versions?pivots=deployment-language-arm-template. This is the ARM template but iirc the same restrictions apply 
```
Character limit: 32-bit integer

Valid characters:
Numbers and periods.
(Each segment is converted to an int32. So each segment has a max value of 2,147,483,647.)
```

Reply:
> yes it did - looking back at this comment: https://github.com/openshift/installer/pull/6304#discussion_r968799213, it did not like latest when used as a version

It doesn't like `latest` when creating an image gallery (version), but it is ok when specifying the image to be used.